dimod/binary_quadratic_model.py

```diff
@@ -2195,3 +2195,33 @@ def from_pandas_dataframe(cls, bqm_df, offset=0.0, interactions=None):
             bqm.add_interaction(u, v, 0.0)
 
         return bqm
+
+
+class OrderedBinaryQuadraticModel(BinaryQuadraticModel):
+    """Consistently ordered varient of :class:`BinaryQuadraticModel`.
```

Comment:
variant

Also, mention how are they ordered -- by order of insertion. Which is well defined if adding variables/interactions one by one (`.add_variable`), but not if constructed from an unordered mapping provided for `linear`/`quadratic` to init.